src/test/java/io/github/intoto/utilities/provenance02/TestEnvelopeGenerator.java

```diff
@@ -0,0 +1,113 @@
+    // Prepare the Builder
+    Builder builder = new Builder();
+    builder.setId("mailto:person@example.com");
+    // Prepare the Invocation
+    Invocation invocation = new Invocation();
+    ConfigSource configSource=new ConfigSource();
+    configSource.setUri("https://example.com/example-1.2.3.tar.gz");
+    configSource.setDigest(Map.of("sha256","323d323edvgd"));
+    configSource.setEntryPoint("src:foo");
+    invocation.setConfigSource(configSource);
+    // Prepare the Materials
+    Material material = new Material();
+    material.setUri("https://example.com/example-1.2.3.tar.gz");
+    material.setDigest(Map.of("sha256", "1234..."));
+    // Prepare Metadata
+    Metadata metadata = new Metadata();
+    metadata.setBuildInvocationId("SomeBuildId");
+    metadata.setBuildStartedOn(OffsetDateTime.parse("1986-12-18T15:20:30+08:00"));
+    metadata.setBuildFinishedOn(OffsetDateTime.parse("1986-12-18T16:20:30+08:00"));
+
+    Completeness completeness = new Completeness();
+    completeness.setParameters(true);
+    completeness.setMaterials(true);
+    completeness.setEnvironment(false);
+    metadata.setCompleteness(completeness);
+
+    // Putting the Provenance together
+    Provenance provenancePredicate = new Provenance();
+    provenancePredicate.setBuilder(builder);
+    provenancePredicate.setBuildType("https://example.com/Makefile");
+    provenancePredicate.setInvocation(invocation);
+    provenancePredicate.setMaterials(List.of(material));
+    provenancePredicate.setMetadata(metadata);
+
+    // ** Putting the Statement together **
+    Statement statement = new Statement();
+    statement.setSubject(List.of(subject));
+    statement.setPredicate(provenancePredicate);
+
+    // Generate a key pair
+    KeyPair keyPair = getKeyPairFromFile();
+    SimpleECDSASigner signer = new SimpleECDSASigner(keyPair.getPrivate(), "MyKey");
+
+    String intotoJsonEnvelope = IntotoHelper.produceIntotoEnvelopeAsJson(statement, signer, false);
+
+    Files.writeString(Path.of(".", "intoto_example.intoto.jsonl"), intotoJsonEnvelope);
+  }
+
+  /**
+   * Gets the keys from the resources directory (public.key and private.key) and loads them up as a
+   * {@link KeyPair}
+   */
+  private static KeyPair getKeyPairFromFile() throws Exception {
+    Security.addProvider(new BouncyCastleProvider());
+    // Getting ClassLoader obj
+    ClassLoader classLoader = TestEnvelopeGenerator.class.getClassLoader();
+
+    // Getting public key
+    File filePublicKey =
+        new File(Objects.requireNonNull(classLoader.getResource("public.pem")).getFile());
+    // Reading with PemReader
+    PublicKey publicKey = readPublicKey(filePublicKey);
+    System.out.println(publicKey.toString());
+
+    // Getting private key
+    File filePrivateKey =
+        new File(Objects.requireNonNull(classLoader.getResource("p8private.pem")).getFile());
+    PrivateKey privateKey = readPrivateKey(filePrivateKey);
+    System.out.println(privateKey.toString());
```

Comment:
We don't want to print. Maybe log instead?

Reply:
+1